packages/back-end/src/api/saved-groups/postSavedGroup.ts

```diff
@@ -9,11 +9,24 @@ import { postSavedGroupValidator } from "../../validators/openapi";
 
 export const postSavedGroup = createApiRequestHandler(postSavedGroupValidator)(
   async (req): Promise<PostSavedGroupResponse> => {
-    if (!req.context.permissions.canCreateSavedGroup()) {
+    const { name, attributeKey, values, condition, owner, projects } = req.body;
+
+    if (!req.context.permissions.canCreateSavedGroup({ ...req.body })) {
       req.context.permissions.throwPermissionError();
     }
 
-    const { name, attributeKey, values, condition, owner } = req.body;
+    if (projects) {
+      await Promise.all(
+        projects.map(async (projectId) => {
+          // Ensure the project exists
+          const project = await req.context.models.projects.getById(projectId);
+          if (!project) {
+            throw new Error("Project does not exist");
+          }
+        })
+      );
```

Comment:
Avoid doing Mongo queries inside of loops since it can be very expensive.  A better approach is to use `getByIds` and pass in an array of ids.

```ts
if (projects && projects.length) {
  const projectObjects = await req.context.models.projects.getByIds(projects);
  if (projectObjects.length !== projects.length) {
    throw new Error("Invalid project ids");
  }
}
```

You can improve the above by having the error message list the specific project ids that are invalid.

And since this validation logic is generally useful (already used in a few places in this PR), it might make sense to add it to ProjectModel directly, so you could just do something like this:

```ts
await req.context.models.projects.ensureProjectsExist(projects);
```